### PR TITLE
Remove `CausalOrdering` from `task-run-recorder`

### PR DESCRIPTION
**Pull request by jakekaplan**
The `task-run-recorder` consumer takes in task run events, converts them into task runs and task run states, and inserts them into the database. Previously we ensured that these inserts are forced to be done in order of the events occurring through the `CausalOrdering`. 

In practice, the only benefit is that if someone were to observe the actual task run state inserts, a `COMPLETED` row would never exist before a `RUNNING` row. Additionally as long as we are using a queue that is always ordered (currently the only option is an in memory queue), events will never come out of order. Additionally the final state of the task run will always be reported correctly.

This PR removes the ordering as there is no real benefit and it is likely adding unnecessary strain to the server.

### CODE REVIEW

File: tests/server/services/test_task_run_recorder.py [hunk 1]
```diff
@@ -711,50 +707,49 @@ async def test_updates_task_run_on_out_of_order_state_change(
...
-    # move time forward so we can record the lost follower
-    with patch("prefect.server.events.ordering.pendulum.now") as the_future:
-        the_future.return_value = now.add(minutes=20)
-        await record_lost_follower_task_run_events()
-
-    assert record_task_run_event_mock.await_count == 1
-    record_task_run_event_mock.assert_awaited_with(event)
+@pytest.mark.parametrize(
+    "event_order",
+    list(permutations(["PENDING", "RUNNING", "COMPLETED"])),
```
Comment (chrisguidry) [inline tests/server/services/test_task_run_recorder.py:712]: :fire:

File: tests/server/services/test_task_run_recorder.py [hunk 2]
```diff
@@ -711,50 +707,49 @@ async def test_updates_task_run_on_out_of_order_state_change(
...
+    ids=lambda x: "->".join(x),
```
Comment (chrisguidry) [inline tests/server/services/test_task_run_recorder.py:713]: :heart:

**Review by chrisguidry — APPROVED** [review]
That test makes this easy to trust!

**Review by cicdw — APPROVED** [review]
niiiiice thank you!

**Comment by cicdw** [comment]
Related to ongoing investigations in https://github.com/PrefectHQ/prefect/issues/15153

**Pull request merged**